src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,193 @@
+    
+    # if there are no aspatial fields in column table then return nothing to table
+    if isone(length(tableinfo))
+      return nothing
+    end
+    # remove `column_name` field from tableinfo to avoid querying the GeoPackage geometry column 
+    deleteat!(tableinfo, findall(x -> isequal(x, row.column_name), tableinfo))
+    columns = join(tableinfo, ", ")
+
+    rowvals = map(DBInterface.execute(db, "SELECT $columns from $tn")) do rv
+      NamedTuple(rv)
+    end
+    rowvals
+  end
+  isnothing(first(tb)) ? first(tb) : vcat(tb...) 
+end
+
+# https://www.geopackage.org/spec/#:~:text=2.1.5.1.2.%20Table%20Data%20Values
+#------------------------------------------------------------------------------
+# # Requirement 21: a gpkg_contents table row with a "features" data_type
+# SHALL contain a gpkg_geometry_columns table
+#
+# Requirement 22: gpkg_geometry_columns table
+# SHALL contain one row record for the geometry column
+# in each vector feature data table
+#
+# Requirement 23: gpkg_geometry_columns table_name column
+# SHALL reference values in the gpkg_contents table_name column
+# for rows with a data_type of 'features'
+#
+# Requirement 24: The column_name column value in a gpkg_geometry_columns row
+# SHALL be the name of a column in the table or view specified by the table_name
+# column value for that row.
+#
+# Requirement 25: The geometry_type_name value in a gpkg_geometry_columns row
+# SHALL be one of the uppercase geometry type names specified
+
+# Requirement 26: The srs_id value in a gpkg_geometry_columns table row
+# SHALL be an srs_id column value from the gpkg_spatial_ref_sys table.
+#
+# Requirement 27: The z value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 28: The m value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 146: The srs_id value in a gpkg_geometry_columns table row
+# SHALL match the srs_id column value from the corresponding row in the
+# gpkg_contents table.
+function gpkggeoms(db, ; layer=1)
+  tb = DBInterface.execute(
+    db,
+    """
+SELECT g.table_name AS tn, g.column_name AS cn, c.srs_id as crs, g.z as elev, srs.organization as org, srs.organization_coordsys_id as org_coordsys_id,
+( SELECT type FROM sqlite_master WHERE lower(name) = lower(c.table_name) AND type IN ('table', 'view')) AS object_type
+FROM gpkg_geometry_columns g, gpkg_spatial_ref_sys srs
+JOIN gpkg_contents c ON ( g.table_name = c.table_name )
+WHERE c.data_type = 'features'
+AND (SELECT type FROM sqlite_master WHERE lower(name) = lower(c.table_name) AND type IN ('table', 'view')) IS NOT NULL
+AND g.srs_id = srs.srs_id
+AND g.srs_id = c.srs_id
+AND g.z IN (0, 1, 2)
+AND g.m = 0
+ LIMIT $layer;
+    """
+  )
+  firstrow = [row for row in first(tb)]
+  # Note: first feature table that is read specifies the CRS to be used on all feature tables resulted from SELECT statement
+
+  srsid, org, orgcoordsysid = firstrow[3], firstrow[5], firstrow[6]
```

Comment:
It is very hard to debug code like this because future maintainers will have no idea of what is stored in `firstrow`, much less `firstrow[3]`, `firstrow[5]`, ...